scanner/src/main/kotlin/scanners/fossid/Utils.kt

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.scanners.fossid
+
+import org.ossreviewtoolkit.clients.fossid.model.rules.IgnoreRule
+import org.ossreviewtoolkit.clients.fossid.model.rules.RuleType
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.PathExclude
+
+private val DIRECTORY_REGEX = "(?<directory>(?:\\w+/)*\\w+)/?(?<starstar>\\*\\*)?".toRegex()
+private val EXTENSION_REGEX = "\\*\\.(?<extension>\\w+)".toRegex()
+private val FILE_REGEX = "(?<file>[\\w.]+)".toRegex()
+
+/**
+ * Convert an ORT path excludes in [excludes] to FossID [IgnoreRule]. If an error is encountered during the mapping, an
+ * issue is added to [issues].
+ *
+ */
+fun convertRules(excludes: Excludes, issues: MutableList<Issue>): List<IgnoreRule> {
+    return excludes.paths.mapNotNull {
+        it.mapToRule().also { mappedRule ->
+            if (mappedRule == null) {
+                issues += Issue(
+                    source = "FossID.convertRules",
+                    message = "Path exclude '${it.pattern}' cannot be converted to an ignore rule.",
+                    severity = Severity.HINT
+                )
+                FossId.logger.warn {
+                    "Path exclude  '${it.pattern}' cannot be converted to an ignore rule."
+                }
+            }
+        }
+    }
+}
+
+private fun PathExclude.mapToRule(): IgnoreRule? {
+    DIRECTORY_REGEX.matchEntire(pattern)?.let { directoryMatch ->
+        val directory = directoryMatch.groups["directory"]!!.value
+        val starStar = directoryMatch.groups["starstar"]?.value
+
+        return if (starStar == null) {
+            IgnoreRule(-1, RuleType.DIRECTORY, directory, -1, "")
+        } else {
+            IgnoreRule(-1, RuleType.DIRECTORY, "$directory/**", -1, "")
+        }
+    }
+
+    EXTENSION_REGEX.matchEntire(pattern)?.let { extensionMatch ->
+        val extension = extensionMatch.groups["extension"]!!.value
+        return IgnoreRule(-1, RuleType.EXTENSION, ".$extension", -1, "")
+    }
+
+    FILE_REGEX.matchEntire(pattern)?.let { fileMatch ->
+        val file = fileMatch.groups["file"]!!.value
+        return IgnoreRule(-1, RuleType.FILE, file, -1, "")
+    }
+
+    return null
+}
+
+/**
+ * Check if some elements of [rulesToTest] are legacy rules i.e. are not present in a reference list (current object).
+ * Create an issue on [issues] for each legacy rule and return a list of the latter.
+ */
+fun List<IgnoreRule>.filterLegacyRules(rulesToTest: List<IgnoreRule>, issues: MutableList<Issue>): List<IgnoreRule> =
```

Comment:
I think this function could be internal.

Reply:
I changed the visibility of `convertRules` too.